test/functional/apps/visualize/_tsvb_chart.ts

```diff
@@ -146,5 +147,79 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         expect(newValue).to.eql('10');
       });
     });
+
+    describe('browser history changes', () => {
+      it('should activate previous/next chart tab and panel config', async () => {
+        await PageObjects.visualBuilder.resetPage();
```

Comment:
nit: it's a really good test, and cover all possible cases, but maybe we can simplify it and test only one base scenario. I'm afraid that large test cases increase the probability that this test will hang somewhere in the middle

Reply:
I'm not exactly sure what do you mean by "probability of hang somewhere in the middle".
I would save it as it is, since we have to handle both browser "back" and "forward" steps.
@dmlemeshko what do you think?

Reply:
@dmlemeshko please review that test